pkgs/by-name/fi/finamp/package.nix

```diff
@@ -0,0 +1,69 @@
+{ lib
+, flutter
+, mpv
+, patchelf
+, fetchFromGitHub
+, makeDesktopItem
+}:
+let
+  version = "0.9.8-beta";
+in
+flutter.buildFlutterApplication {
+  inherit version;
+  pname = "finamp";
+  src = fetchFromGitHub {
+    owner = "jmshrv";
+    repo = "finamp";
+    rev = version;
+    hash = "sha256-lvjhA+hdCXgDsrNhNw4Tiq6ZgkYlPuMeHha8OJNF1TI=";
+  };
+  pubspecLock = lib.importJSON ./pubspec.lock.json;
+
+  nativeBuildInputs = [ patchelf ];
+  buildInputs = [ mpv ];
+
+  gitHashes = {
+    balanced_text = "sha256-lSDR5dDjZ4garRbBPI+wSxC5iScg8wVSD5kymmLbYbk=";
+    isar_generator = "sha256-lWnHmZmYx7qDG6mzyDqYt+Xude2xVOH1VW+BoDCas60=";
+    media_kit_libs_windows_audio = "sha256-p3hRq79whLFJLNUgL9atXyTGvOIqCbTRKVk1ie0Euqs=";
+    palette_generator = "sha256-mnRJf3asu1mm9HYU8U0di+qRk3SpNFwN3S5QxChpIA0=";
+    split_view = "sha256-unTJQDXUUPVDudlk0ReOPNYrsyEpbd/UMg1tHZsmg+k=";
+  };
+
+  postFixup = ''
+    patchelf $out/app/finamp --add-needed libisar.so --add-needed libmpv.so --add-rpath ${lib.makeLibraryPath [ mpv ]}
+  '';
+
+  postInstall = ''
+    d=$out/share/icons/hicolor/scalable/apps
+    mkdir -p $d
+    cp $src/assets/icon/icon_foreground.svg $d/finamp.svg
+
+    mkdir -p $out/share/applications
+    cp $desktopItem/share/applications/*.desktop $out/share/applications
+  '';
+
+  desktopItem = makeDesktopItem {
+    name = "Finamp";
+    desktopName = "Finamp";
+    genericName = "Music Player";
+    exec = "finamp";
+    icon = "finamp";
+    startupWMClass = "finamp";
+    comment = "An open source Jellyfin music player";
+    categories = [
+      "Audio"
```

Comment:
Just noticed this while building:

```
/nix/store/5ddf158z5769lbw9biq97npk0sxfd061-Finamp.desktop/share/applications/Finamp.desktop: error: (will be fatal in the future): value item "Audio" in key "Categories" in group "Desktop Entry" requires another category to be present among the following categories: AudioVideo
```

Adding `AudioVideo` to the categories should fix it.

Reply:
I added `AudioVideo` to the categories.
[Flutter first tries `/etc/os-release` and falls back to `/usr/lib/os-release` if it doesn't exist.](https://github.com/flutter/flutter/blob/5187cab7bdd434ca74abb45895d17e9fa553678a/packages/flutter_tools/lib/src/base/os.dart#L321)
I noticed that during build `/etc` only contains `group`, `hosts` and `passwd`.

Reply:
Oh yes, that may be because builds are sandboxed.